### ISSUE TITLE
Incorrect access of global when used in a worker

### ISSUE THREAD

**Issue by cdata** [devashishdxt/rexie#7]
Hey there, thanks for building this great library.

I've been doing some Rust/WASM-in-a-Web-Worker stuff lately, and I've noticed a few libraries that could be useful in workers are accessing the global scope in a way that makes it impossible to use them there.

When running in a standard Web Worker, the global changes from `Window` to `DedicatedWorkerGlobalScope`, and the constructor for `Window` is no longer defined. This means any attempt to access it will result in an error. Similarly, the `DedicatedWorkerGlobalScope` constructor doesn't exist in the main document, so any attempt to access _that_ will fail if not done in a worker.

Rexie tries to access `Window` regardless of whether it is running in the main thread. I've created a patch that fixes this problem as an example of how to make Rexie work in more contexts: https://github.com/devashishdxt/rexie/compare/main...cdata:main

I'm currently able to use Rexie in a worker with the above patch. I would be happy to turn the patch into a pull request if that is something you would like to consider.

**Comment by devashishdxt** [comment]
Hi @cdata. Thanks for creating this issue.

It'd be great if you can create a PR for this change. I'm not that familiar with JavaScript and especially not with web workers.

It'd be great if you can also add some tests for using this in web workers.

Thanks.

**Comment by devashishdxt** [comment]
Please refer this to run the tests: https://rustwasm.github.io/docs/wasm-bindgen/wasm-bindgen-test/browsers.html

**Comment by cdata** [comment]
@devashishdxt thanks! I will try to add some worker tests. I'm familiar with `wasm-bindgen-test` as I have recently been using it quite a lot.

I will advise: at this time it may be necessary to run to worker tests via a separate target that gets built in advance and loaded at test time. This is because `wasm-bindgen-test` runs its own `wasm-bindgen` invocation on the test target and hard-codes the configuration:

https://github.com/rustwasm/wasm-bindgen/blob/e4477293c15df8500330b295cb9624dccb2a063b/crates/cli/src/bin/wasm-bindgen-test-runner/main.rs#L152

The consequence of this is that you cannot build targets with the [`--target no-modules`](https://rustwasm.github.io/wasm-bindgen/examples/without-a-bundler.html#using-the-older---target-no-modules) configuration. However, this configuration is necessary in order to generate output from `wasm-bindgen` that can be directly loaded in a worker:

https://rustwasm.github.io/wasm-bindgen/examples/wasm-in-web-worker.html#building--compatibility

In summary, `wasm-bindgen-test-runner` cannot directly product output that runs in a Web Worker.

**Comment by devashishdxt** [comment]
Thanks @cdata for the explanation. As I mentioned earlier, I'm not that familiar with web workers and how to use them. My main focus for this crate is easy-of-use. So, if we're planning to add support for web-workers in this crate, it should be easy for users to use it. 

I'd be very helpful if you can create a branch on your fork with some working code so that I can see how everything fits together.

**Comment by cdata** [comment]
Understood. I'm only mentioning this because it will add complexity to how the tests are implemented. The patch I linked above (https://github.com/devashishdxt/rexie/compare/main...cdata:main) works and does not change how difficult it is to use Rexie, it just makes it possible to use it in a worker.

**Comment by devashishdxt** [comment]
@cdata Thanks for clarification.
